src/codewhisperer/service/inlineCompletionService.ts

```diff
@@ -142,42 +150,105 @@ export class InlineCompletionService {
         TelemetryHelper.instance.tryRecordClientComponentLatency()
     }
 
-    setCodeWhispererStatusBarLoading() {
-        this._isPaginationRunning = true
-        const selectedCustomization = getSelectedCustomization()
-        this.statusBar.text = codicon` ${getIcon('vscode-loading~spin')} CodeWhisperer${
-            selectedCustomization.arn === '' ? '' : ` | ${selectedCustomization.name}`
-        }`
-        this.statusBar.command = undefined
-        ;(this.statusBar as any).backgroundColor = undefined
-        this.statusBar.show()
-    }
-
-    setCodeWhispererStatusBarOk() {
-        this._isPaginationRunning = false
-        const selectedCustomization = getSelectedCustomization()
-        this.statusBar.text = codicon`${getIcon('vscode-check')} CodeWhisperer${
-            selectedCustomization.arn === '' ? '' : ` | ${selectedCustomization.name}`
-        }`
-        this.statusBar.command = undefined
-        ;(this.statusBar as any).backgroundColor = undefined
-        this.statusBar.show()
+    /** Updates the status bar to represent the latest CW state */
+    refreshStatusBar() {
+        if (AuthUtil.instance.isConnectionValid()) {
+            return this.setState('ok')
+        } else if (AuthUtil.instance.isConnectionExpired()) {
+            return this.setState('expired')
+        } else {
+            return this.setState('notConnected')
+        }
     }
 
-    setCodeWhispererStatusBarDisconnected() {
+    private async setState(state: keyof typeof states) {
         this._isPaginationRunning = false
-        this.statusBar.text = codicon` ${getIcon('vscode-debug-disconnect')} CodeWhisperer`
-        this.statusBar.command = 'aws.codeWhisperer.reconnect'
-        ;(this.statusBar as any).backgroundColor = new vscode.ThemeColor('statusBarItem.warningBackground')
-        this.statusBar.show()
+        switch (state) {
+            case 'loading': {
+                this._isPaginationRunning = true
+                await this.statusBar.setState('loading')
+                break
+            }
+            case 'ok': {
+                await this.statusBar.setState('ok', CodeSuggestionsState.instance.isSuggestionsEnabled())
+                break
+            }
+            case 'expired': {
+                await this.statusBar.setState('expired')
+                break
+            }
+            case 'notConnected': {
+                await this.statusBar.setState('notConnected')
+                break
+            }
+        }
     }
 
     isPaginationRunning(): boolean {
         return this._isPaginationRunning
     }
+}
 
-    hideCodeWhispererStatusBar() {
-        this._isPaginationRunning = false
-        this.statusBar.hide()
+/** The states that the completion service can be in */
+const states = {
+    loading: 'loading',
+    ok: 'ok',
+    expired: 'expired',
+    notConnected: 'notConnected',
+} as const
+
+export class CodeWhispererStatusBar {
```

Comment:
would it be more discoverable for this to live in a module next to `auth/ui/statusBarItem.ts` ? `inlineCompletionService.ts` seems like a random place for it to live.